test/DynamoCoreTests/Settings.cs

```diff
@@ -84,6 +85,31 @@ public void SetPythonTemplateFromConfigWithInvalidPath()
             Assert.AreEqual(model.PreferenceSettings.PythonTemplateFilePath, string.Empty);
         }
 
-    }
+        [Test]
+        public void RunSettingsDisableAndEnableRun()
+        {
+            string openPath = Path.Combine(TestDirectory, @"core\RunSettings.dyn");
+            CurrentDynamoModel.OpenFileFromPath(openPath);
+
+            var ws = CurrentDynamoModel.CurrentWorkspace as HomeWorkspaceModel;
+            Assert.AreEqual(0, ws.EvaluationCount);
+
+            // Still in Manual mode so will not run
+            ws.RequestRun();
+            Assert.AreEqual(0, ws.EvaluationCount);
 
+            // Setting the run type to automatic
+            ws.RunSettings.RunType = RunType.Automatic;
+
+            ws.RunSettings.RunEnabled = false;
+            // This should not run
+            ws.RequestRun();
+            Assert.AreEqual(0, ws.EvaluationCount);
+
+            ws.RunSettings.RunEnabled = true;
+            // This should run
+            ws.RequestRun();
+            Assert.AreEqual(1, ws.EvaluationCount);
+        }
+    }
```

Comment:
would be better if we can have a test for connecting nodes on an automatic graph won't run the graph if run is disabled.

Reply:
Yes, modified the test to verify this.